woocommerce/woocommerce-android
# Use suspendable fetchSingleOrder

**We'll need to update fluxc hash before we merge this PR.**




Parent issue https://github.com/woocommerce/woocommerce-android/issues/4806

### Description


"FetchSingleOrder" action was refactored into a suspendable method - more info on FluxC PR https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2118.

This PR is the first of 7/8 PRs - their goal is to transform OrderDetailRepository into a stateless singleton. The only state the repository holds are continuations which won't be necessary as soon as all the methods are using suspendable functions instead of event bus.

### Testing instructions


1. Open detail of a paid order
2. Issue a refund
3. Notice when the refund is done, the order gets updated

### Images/gif


https://i.gyazo.com/e1af0723b506cb680740f14862e35e2d.gif
